File: capture.js
const puppeteer = require('puppeteer');
const express = require('express');
const path = require('path');
const fs = require('fs');

const app = express();
const port = 3000;
app.use(express.static(__dirname));

const server = app.listen(port, async () => {
  console.log(`Server running at http://localhost:${port}`);
  
  const browser = await puppeteer.launch({ headless: true });
  const page = await browser.newPage();
  await page.setViewport({ width: 1280, height: 720 });
  
  const projects = [
    { url: `http://localhost:${port}/CIPT/index.html`, name: 'cipt-screen.png' },
    { url: `http://localhost:${port}/carbon-footprint-platform-main/dist/index.html`, name: 'carbon-screen.png' },
    { url: `http://localhost:${port}/Carbon-Footprint-Results-main/Carbon-Footprint-Results-main/index.html`, name: 'results-screen.png' },
    { url: `http://localhost:${port}/flashcards-game-main/index.html`, name: 'flashcards-screen.png' },
    { url: `http://localhost:${port}/Altahera-Management-System/index.html`, name: 'altahera-screen.png' },
    { url: `http://localhost:${port}/radiance-ris-main/dist/index.html`, name: 'radiance-screen.png' },
  ];

  if (!fs.existsSync(path.join(__dirname, 'assets', 'screenshots'))) {
    fs.mkdirSync(path.join(__dirname, 'assets', 'screenshots'), { recursive: true });
  }

  for (const project of projects) {
    try {
      console.log(`Capturing ${project.name}...`);
      await page.goto(project.url, { waitUntil: 'networkidle0', timeout: 30000 });
      // wait a bit for animations
      await new Promise(r => setTimeout(r, 2000));
      await page.screenshot({ path: path.join(__dirname, 'assets', 'screenshots', project.name) });
      console.log(`Saved ${project.name}`);
    } catch (e) {
      console.error(`Failed to capture ${project.name}: ${e.message}`);
    }
  }

  await browser.close();
  server.close();
});

File: capture.py
import time
import os
from selenium import webdriver
from selenium.webdriver.edge.options import Options

def capture_screenshots():
    options = Options()
    options.add_argument('--headless')
    options.add_argument('--window-size=1280,720')
    options.add_argument('--hide-scrollbars')
    
    driver = webdriver.Edge(options=options)
    
    os.makedirs('assets/screenshots', exist_ok=True)
    
    projects = [
        {'url': 'http://localhost:8080/CIPT/index.html', 'name': 'cipt-screen.png'},
        {'url': 'http://localhost:8080/carbon-footprint-platform-main/dist/index.html', 'name': 'carbon-screen.png'},
        {'url': 'http://localhost:8080/Carbon-Footprint-Results-main/Carbon-Footprint-Results-main/index.html', 'name': 'results-screen.png'},
        {'url': 'http://localhost:8080/flashcards-game-main/index.html', 'name': 'flashcards-screen.png'},
        {'url': 'http://localhost:8080/Altahera-Management-System/index.html', 'name': 'altahera-screen.png'},
        {'url': 'http://localhost:8080/radiance-ris-main/dist/index.html', 'name': 'radiance-screen.png'},
    ]
    
    for p in projects:
        print(f"Capturing {p['name']}...")
        driver.get(p['url'])
        time.sleep(2)  # wait for animations
        driver.save_screenshot(f"assets/screenshots/{p['name']}")
        print(f"Saved {p['name']}")
        
    driver.quit()

if __name__ == '__main__':
    capture_screenshots()
